### ISSUE TITLE
NTI fails to catch error "Access to private property not allowed"

### ISSUE THREAD

**Issue by myphysicslab** [google/closure-compiler#2416]
Without NTI I get this proper error:

```
src/sims/engine2D/CarSuspensionApp.js:228: ERROR - Access to private property proto_ of 
myphysicslab.lab.view.DisplayShape not allowed here.
      this.displayList.findShape(wheel1).proto_ = this.protoWheel;
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

But with NTI that error does not appear.  Here are the NTI options I'm using:

```
--new_type_inf \
--jscomp_warning=newCheckTypes \
--jscomp_off=newCheckTypesExtraChecks \
--hide_warnings_for=`readlink closure-library` \
```
After removing those options the error appears.

Here is the complete compile command:

```
java -jar ../closure-compiler/target/closure-compiler-1.0-SNAPSHOT.jar
--entry_point=goog:myphysicslab.sims.engine2D.CarSuspensionApp
--compilation_level=SIMPLE --define=goog.DEBUG=false '--define=goog.LOCALE='\''en'\'''
--define=myphysicslab.lab.util.UtilityCore.ADVANCED=false --generate_exports
--js=../closure-library/ --js=src --jscomp_error=accessControls
--jscomp_error=ambiguousFunctionDecl --jscomp_error=checkTypes --jscomp_error=checkVars
--jscomp_error=const --jscomp_error=constantProperty --jscomp_error=fileoverviewTags
--jscomp_error=globalThis --jscomp_error=invalidCasts
--jscomp_error=misplacedTypeAnnotation --jscomp_error=missingProperties
--jscomp_error=missingProvide --jscomp_error=missingRequire
--jscomp_error=missingReturn --jscomp_error=strictModuleDepCheck
--jscomp_error=suspiciousCode --jscomp_error=typeInvalidation
--jscomp_error=undefinedNames --jscomp_error=undefinedVars
--jscomp_error=unknownDefines --jscomp_error=uselessCode --jscomp_error=visibility
--new_type_inf --jscomp_warning=newCheckTypes --jscomp_off=newCheckTypesExtraChecks
--hide_warnings_for=../closure-library/ --emit_use_strict
--language_in=ECMASCRIPT5_STRICT --language_out=ECMASCRIPT5_STRICT
--dependency_mode=STRICT --warning_level=VERBOSE
'--output_wrapper=/*Copyright_2016_Erik_Neumann_All_Rights_Reserved_www.myphysicslab.com*/%output%'
```

Here is the definition of `findShape`

```javascript
/** Returns the DisplayShape that shows the given SimObject.
@param {!SimObject|string|number} search  the SimObject to search for, or name of
    SimObject, or index number of DisplayObject.
    Name should be English or language-independent version of name.
@return {?DisplayShape} the DisplayShape on this list that shows
    the given SimObject, or null if not found
*/
DisplayList.prototype.findShape = function(search) {
  var ds = this.find(search);
  if (ds instanceof DisplayShape) {
    return /**!DisplayShape*/(ds);
  }
  return null;
};
```

Here is part of the definition of `DisplayShape`

```javascript
/**
* @param {?myphysicslab.lab.model.MassObject=} massObject the MassObject to display
* @param {?DisplayShape=} proto the prototype DisplayShape to inherit properties
*     from
* @constructor
* @final
* @struct
* @implements {myphysicslab.lab.view.DisplayObject}
*/
myphysicslab.lab.view.DisplayShape = function(massObject, proto) {
  /** The MassObject to display.
  * @type {!myphysicslab.lab.model.MassObject}
  * @private
  */
  this.massObject_ = goog.isDefAndNotNull(massObject) ?
      massObject : new PointMass('proto');
  /**
  * @type {?DisplayShape}
  * @private
  */
  this.proto_ = goog.isDefAndNotNull(proto) ? proto : null;
```

**Comment by brad4d** [comment]
@myphysicslab 
There's not quite enough info in your post to see what's happening here.
Could you create a reproduction using http://closure-compiler.appspot.com/home
and post it here?
Thanks.

**Comment by myphysicslab** [comment]
This is also solved by what I show in my recent comment on #2415. After making that change I get the correct error here using NTI:

```
src/sims/engine2D/CarSuspensionApp.js:232: ERROR - Access to private property proto_ of 
myphysicslab.lab.view.DisplayShape not allowed here.
      this.displayList.findShape(wheel1).proto_ = this.protoWheel;
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by dimvar** [comment]
Just replied on the other issue.
